### PR TITLE
DCOS_OSS-5412: add diagnostic service

### PR DESCRIPTION
**Pull request by OleksandrBielov**
@sebbrandt87, @fatz  , please review and approve commit

### CODE REVIEW

File: roles/DCOS.agent_windows/tasks/main.yml [hunk 1]
```diff
@@ -47,3 +47,37 @@
...
     start_mode: auto
   notify: restart the service
+
+- name: Set an environment variable for all users
+  win_environment:
+    state: present
+    name: DCOS_VERSION
+    value: "{{ dcos['version'] }}"
+    level: machine
+
```
Comment (sebbrandt87) [inline roles/DCOS.agent_windows/tasks/main.yml]: extra empty line

Comment (sebbrandt87) [inline roles/DCOS.agent_windows/tasks/main.yml:57]: extra empty line?

File: roles/DCOS.agent_windows/tasks/main.yml [hunk 2]
```diff
@@ -47,3 +47,37 @@
     dependencies: [ 'docker' ]
     start_mode: auto
   notify: restart the service
+
+- name: Set an environment variable for all users
```
Comment (sebbrandt87) [inline roles/DCOS.agent_windows/tasks/main.yml]: please add `Windows | `

File: roles/DCOS.agent_windows/tasks/main.yml [hunk 3]
```diff
@@ -47,3 +47,37 @@
...
+  win_environment:
+    state: present
+    name: DCOS_VERSION
+    value: "{{ dcos['version'] }}"
+    level: machine
+
+
+- name: Check that dcos-diagnostics.exe exists
```
Comment (sebbrandt87) [inline roles/DCOS.agent_windows/tasks/main.yml]: please add `Windows | `

File: roles/DCOS.agent_windows/tasks/main.yml [hunk 4]
```diff
@@ -47,3 +47,37 @@
...
+    url: "{{ url_to_diagnostics }}"
+    dest: C:\dcos\dcos-diagnostics.zip
+    force: no
+  when: stat_result.stat.exists == False 
+
+- name: Windows | Unzip the dcos-diagnostics
+  win_unzip:
+    src: C:\dcos\dcos-diagnostics.zip
+    dest: C:\dcos\
+  when: stat_result.stat.exists == False
```
Comment (sebbrandt87) [inline roles/DCOS.agent_windows/tasks/main.yml]: `when: not stat_result.stat.exists`

File: roles/DCOS.agent_windows/tasks/main.yml [hunk 5]
```diff
@@ -47,3 +47,37 @@
...
+  win_stat:
+    path: C:\dcos\dcos-diagnostics.exe
+  register: stat_result
+
+- name: Windows | Download dcos-diagnostics
+  win_get_url:
+    url: "{{ url_to_diagnostics }}"
+    dest: C:\dcos\dcos-diagnostics.zip
+    force: no
+  when: stat_result.stat.exists == False
```
Comment (sebbrandt87) [inline roles/DCOS.agent_windows/tasks/main.yml]: `when: not stat_result.stat.exists`

**Review by sebbrandt87 — CHANGES_REQUESTED** [review]

File: roles/DCOS.agent_windows/tasks/main.yml (hunk 1 above)
Comment (OleksandrBielov) [reply roles/DCOS.agent_windows/tasks/main.yml:57]: fixed at 0904cc8bac0eca6eaa2fa962d5178c5a3f697580

File: roles/DCOS.agent_windows/tasks/main.yml (hunk 2 above)
Comment (OleksandrBielov) [reply roles/DCOS.agent_windows/tasks/main.yml]: fixed at 0904cc8bac0eca6eaa2fa962d5178c5a3f697580

File: roles/DCOS.agent_windows/tasks/main.yml (hunk 3 above)
Comment (OleksandrBielov) [reply roles/DCOS.agent_windows/tasks/main.yml]: fixed at 0904cc8bac0eca6eaa2fa962d5178c5a3f697580

File: roles/DCOS.agent_windows/tasks/main.yml (hunk 4 above)
Comment (OleksandrBielov) [reply roles/DCOS.agent_windows/tasks/main.yml]: fixed at 0904cc8bac0eca6eaa2fa962d5178c5a3f697580

File: roles/DCOS.agent_windows/tasks/main.yml (hunk 5 above)
Comment (OleksandrBielov) [reply roles/DCOS.agent_windows/tasks/main.yml]: fixed at 0904cc8bac0eca6eaa2fa962d5178c5a3f697580

File: roles/DCOS.agent_windows/tasks/main.yml [hunk 6]
```diff
@@ -47,3 +47,36 @@
...
+    dest: C:\dcos\
+  when: not stat_result.stat.exists
+
+- name: Windows | Install Diagnostic Service
+  win_nssm:
+    name: "{{ mesos_workers.windows.diagnostics_service_name }}"
+    application: "{{ mesos_workers.windows.diagnostics_executor }}"
+    app_parameters_free_form: --config "{{ mesos_workers.windows.diagnostics_config }}" --role agent daemon
+  notify: restart dcos-diagnostics service
+
```
Comment (sebbrandt87) [inline roles/DCOS.agent_windows/tasks/main.yml]: lint is complaining about this empty line

**Review by sebbrandt87 — CHANGES_REQUESTED** [review]

File: roles/DCOS.agent_windows/tasks/main.yml (hunk 6 above)
Comment (OleksandrBielov) [reply roles/DCOS.agent_windows/tasks/main.yml]: fixed at ea04c49d9d011f9b1efe4dd8242759c4e14d29ec

**Review by sebbrandt87 — APPROVED** [review]

**Pull request merged**